### PR TITLE
`Development`: Improve JPQL style for competency repositories

### PR DESCRIPTION
**Pull request by julian-christl**
#### General

- [X] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [X] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).

### Motivation and Context

I just fixed the style in one repository so I decided to go through some more repos to make the codebase more consistent and clean.
Extracted an inner-class enum to work properly with Hibernate.

### Description

I enforced capitalisation (SELECT instead of select), simple variable declaration (:var instead of :#{#var}, explicit enum values, consistent newlines and indentations

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### Test Coverage

No functional changes.

## Summary by CodeRabbit

- **Refactor**
	- Improved query formatting and method signatures in CompetencyProgressRepository, CompetencyRelationRepository, CompetencyRepository, and LearningPathRepository for better performance and readability.
	- Standardized SQL query syntax and comparison operators across CompetencyRelationRepository and CompetencyRepository.
	- Enhanced query conditions in LearningPathRepository for more precise filtering based on course and user attributes.

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/repository/CompetencyProgressRepository.java [hunk 1]
```diff
@@ -17,7 +17,10 @@ public interface CompetencyProgressRepository extends JpaRepository<CompetencyPr
 
     @Transactional // ok because of delete
     @Modifying
-    @Query("DELETE FROM CompetencyProgress cp WHERE cp.learningGoal.id = :learningGoalId")
+    @Query("""
+            DELETE FROM CompetencyProgress cp
+            WHERE cp.learningGoal.id = :learningGoalId
+            """)
     void deleteAllByCompetencyId(Long learningGoalId);
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/repository/CompetencyProgressRepository.java:24]: `@Param`

**Review by nityanandaz — DISMISSED** [review]
Code review: looks good

**Review by Strohgelaender — DISMISSED** [review]

**Review by rstief — APPROVED** [review]
code looks good

File: src/main/java/de/tum/in/www1/artemis/repository/CompetencyProgressRepository.java [hunk 2]
```diff
@@ -71,23 +69,23 @@ SELECT AVG(cp.confidence)
...
     @Query("""
-            SELECT count(cp)
+            SELECT COUNT(cp)
             FROM CompetencyProgress cp
             WHERE cp.competency.id = :competencyId
                 AND cp.progress >= :progress
                 AND cp.confidence >= :confidence
             """)
-    Long countByCompetencyAndProgressAndConfidenceGreaterThanEqual(@Param("competencyId") Long competencyId, @Param("progress") Double progress,
+    Long countByCompetencyAndProgressAndConfidenceGreaterThanEqual(@Param("competencyId") long competencyId, @Param("progress") Double progress,
```
Comment (MaximilianAnzinger) [inline src/main/java/de/tum/in/www1/artemis/repository/CompetencyProgressRepository.java:88]:
```suggestion
    Long countByCompetencyAndProgressAndConfidenceGreaterThanEqual(@Param("competencyId") long competencyId, @Param("progress") double progress,
```

File: src/main/java/de/tum/in/www1/artemis/repository/CompetencyProgressRepository.java [hunk 3]
```diff
@@ -71,23 +69,23 @@ SELECT AVG(cp.confidence)
...
             @Param("confidence") Double confidence);
```
Comment (MaximilianAnzinger) [inline src/main/java/de/tum/in/www1/artemis/repository/CompetencyProgressRepository.java:89]:
```suggestion
            @Param("confidence") double confidence);
```

**Review by MaximilianAnzinger — CHANGES_REQUESTED** [review]

**Review by bassner — APPROVED** [review]

**Review by MaximilianAnzinger — APPROVED** [review]

**Pull request merged**